Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift

```diff
@@ -5,169 +5,92 @@
+
     // MARK: - Property
 
-    public let uiConfig: UIConfig<ExtraData>
-    
-    public var suggestions: [String] = ["Brian", "David", "Pavel"] {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var position: CGPoint = CGPoint(x: 0, y: 300) {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var maxHeight: CGFloat = 300.0 {
-        didSet {
-            updateContent()
-        }
-    }
-    
-    public var calculatedHeight: CGFloat {
-        min(CGFloat(40 * suggestions.count), maxHeight)
-    }
-    
+    private var collectionViewHeightObserver: NSKeyValueObservation?
+
     // MARK: - Subviews
-    
-    public private(set) lazy var tableView = UITableView().withoutAutoresizingMaskConstraints
-    
-    // MARK: - Init
-    
-    public required init(
-        uiConfig: UIConfig<ExtraData> = .default
-    ) {
-        self.uiConfig = uiConfig
-
-        super.init(nibName: nil, bundle: nil)
-        
-        commonInit()
-    }
-    
-    public required init?(coder: NSCoder) {
-        uiConfig = .default
-        
-        super.init(coder: coder)
-        
-        commonInit()
-    }
-    
-    public func commonInit() {
-        tableView.dataSource = self
-        tableView.delegate = self
-        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
-        modalPresentationStyle = .custom
-        modalTransitionStyle = .crossDissolve
-    }
-    
+
+    public private(set) lazy var collectionView = uiConfig
+        .messageComposer
+        .suggestionsCollectionView
+        .init(layout: uiConfig.messageComposer.suggestionsCollectionViewLayout.init())
+        .withoutAutoresizingMaskConstraints
+
     // MARK: - Overrides
-    
+
     override open func viewDidLoad() {
         super.viewDidLoad()
-        
-        view.embed(tableView)
-        
-        setupAppearance()
-        setupLayout()
+        view.embed(collectionView)
     }
-    
-    override open func updateViewConstraints() {
-        widthConstraint?.constant = UIScreen.main.bounds.width
-        heightConstraint?.constant = calculatedHeight
-        
-        positionXConstraint?.constant = position.x
-        positionYConstraint?.constant = position.y
-        
-        super.updateViewConstraints()
+
+    override open func setUp() {
+        super.setUp()
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: "Cell")
+
+        collectionViewHeightObserver = collectionView.observe(
+            \.contentSize,
+            options: [.new],
+            changeHandler: { [weak self] _, change in
+                guard let self = self, let newSize = change.newValue else { return }
+                self.heightConstraint?.constant = newSize.height
+                self.view.layoutIfNeeded()
+            }
+        )
```

Comment:
This needs to have some upper limit from which the content of the collection view should start scrolling.